### PR TITLE
Move stage 3 role creation to resman

### PR DESCRIPTION
**Pull request by juliocc**

### CODE REVIEW

File: fast/stages/1-resman/stage-2-networking.tf [hunk 1]
```diff
@@ -83,10 +83,10 @@ module "net-folder" {
     },
     # project factory service accounts
     (var.fast_stage_2.project_factory.enabled) != true ? {} : {
-      (var.custom_roles.service_project_network_admin) = [
+      (var.custom_roles["service_project_network_admin"]) = [
```
Comment (wiktorn) [inline fast/stages/1-resman/stage-2-networking.tf:86]:
```suggestion
      (local.custom_roles["service_project_network_admin"]) = [
```

File: fast/stages/1-resman/stage-2-networking.tf [hunk 2]
```diff
@@ -83,10 +83,10 @@ module "net-folder" {
...
         module.pf-sa-rw[0].iam_email
       ]
-      (var.custom_roles.project_iam_viewer) = [
+      (var.custom_roles["project_iam_viewer"]) = [
```
Comment (wiktorn) [inline fast/stages/1-resman/stage-2-networking.tf:89]:
```suggestion
      (local.custom_roles["project_iam_viewer"]) = [
```

File: fast/stages/1-resman/moved/v35.1.0-v36.0.0.tf [hunk 3]
```diff
@@ -236,3 +236,21 @@ moved {
...
+  for_each = var.fast_stage_2.network_security.enabled ? [""] : []
+  to       = module.nsec-organization-roles[0].google_organization_iam_custom_role.roles["ngfw_enterprise_admin"]
+  id       = "organizations/${var.organization.id}/roles/ngfwEnterpriseAdmin"
+}
+
+import {
+  for_each = var.fast_stage_2.network_security.enabled ? [""] : []
+  to       = module.nsec-organization-roles[0].google_organization_iam_custom_role.roles["ngfw_enterprise_viewer"]
+  id       = "organizations/${var.organization.id}/roles/ngfwEnterpriseViewer"
+}
```
Comment (wiktorn) [inline fast/stages/1-resman/moved/v35.1.0-v36.0.0.tf:256]: The migration scenario then assumes following actions:
* run `terraform state rm ...` on 0-bootstrap - to remove roles from state and prevent their deletion on `apply`
* run `terraform apply` on 1-resman

I guess, to keep the migration paths clean, we need to call out that manual step somewhere.

**Review by wiktorn — COMMENTED** [review]
Not functional, but just to save myself wondering why sometime we use `local.custom_roles` and sometimes `var.custom_roels` I'd align `custom_roles` access to go through `local` everywhere. There are still left some leftovers in `billing.tf` and `stage*.tf`.

File: fast/stages/1-resman/stage-3.tf [hunk 4]
```diff
@@ -108,6 +111,22 @@ locals {
...
+  stage3_factories = [
+    for k, v in local.stage3 :
+    v.factories_config
+    if v.factories_config != null
+  ]
+}
+
+# Stage 3 custom role factories
+
+module "organization-roles" {
```
Comment (ludoo) [inline fast/stages/1-resman/stage-3.tf:123]: do we really need extra module calls for this? can't we roll up modules in the existing organization module?

**Review by ludoo — APPROVED** [review]
very nice, I would just try reusing the current organization module invocation and bundle up all roles deriving from stage 3 factories in an internal factory local

File: fast/stages/1-resman/moved/v35.1.0-v36.0.0.tf (hunk 3 above)
Comment (juliocc) [reply fast/stages/1-resman/moved/v35.1.0-v36.0.0.tf:256]: Correct. I tried adding a `removed` block  to bootstrap but couldn't figure it out.

**Pull request closed without merging**